drivers/pir/pir.c

```diff
@@ -100,12 +131,30 @@ static void pir_callback(void *arg)
 {
     DEBUG("pir_callback: %p\n", arg);
     pir_t *dev = (pir_t*) arg;
+    bool pin_now = gpio_read(dev->p.gpio);
+    uint64_t now = xtimer_usec_from_ticks64(xtimer_now64());
+
+    /* We were busy counting */
+    if (dev->active && dev->start_active_time != 0xFFFFFFFFFFFFFFFF) {
+        /* Add into accumulation */
+        dev->accum_active_time += (now - dev->start_active_time);
+    }
+    /* Pin is rising */
+    if (pin_now == dev->p.active_high) {
+        dev->start_active_time = now;
+        dev->active = true;
+    /* Pin is falling */
+    } else {
+        dev->start_active_time = 0xFFFFFFFFFFFFFFFF;
```

Comment:
is this marking actuallly needed i.e. isn't the active flag sufficient? After all this callback is only called when the pin level changes, right?

Reply:
I think so too. Thanks!